File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/UaObjectId.java
package org.opcfoundation.webserver.types;

import org.jspecify.annotations.Nullable;
import org.opcfoundation.webserver.addressspace.nodes.UaObject;

public class UaObjectId {
    private final String id;
    private final @Nullable UaObject instance;

    public UaObjectId(String id, @Nullable UaObject instance)
    {
        this.id = id;
        this.instance = instance;
    }

    public String getId() {
        return id;
    }

    public @Nullable UaObject getInstance() {
        return instance;
    }

    @Override
    public String toString() {
        String ret = "Id: " + id;

        if (null != instance)
        {
            ret += " Path: " + instance.browseName();
        }

        return ret;
    }
}

File: opcua-webservice-node/src/server/types/common/UaObjectId.ts
import type { UaObject } from "../../addressspace/nodes/UaObject";

export class UaObjectId {
    private readonly _id: string;
    private readonly _instance: UaObject | null;

    constructor(id: string, instance: UaObject | null) {
        this._id = id;
        this._instance = instance;
    }

    get id(): string {
        return this._id;
    }

    get instance(): UaObject | null {
        return this._instance;
    }

    toString(): string {
        let ret = this._id;

        if (this._instance !== null) {
            ret += `${this._instance.browseName}`;
        }

        return ret;
    }
}